File: src/Cyniris/SyndicateBundle/Entity/State.php
<?php
namespace Cyniris\SyndicateBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity
 * @ORM\Table(name="state")
 */
class State {
    /**
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     * @ORM\Column(type="integer")
     */
    protected $id;

     /**
     * @ORM\Column(type="boolean")
     */
    protected $read;

     /**
     * @ORM\Column(type="boolean")
     */
    protected $stared;

    /**
     * @ORM\ManyToOne(targetEntity="Post")
     */
    protected $post;

    /**
     * @ORM\ManyToOne(targetEntity="User")
     */
    protected $user;

    /**
     * Get id
     *
     * @return integer 
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set read
     *
     * @param boolean $read
     * @return State
     */
    public function setRead($read)
    {
        $this->read = $read;

        return $this;
    }

    /**
     * Get read
     *
     * @return boolean 
     */
    public function getRead()
    {
        return $this->read;
    }

    /**
     * Set stared
     *
     * @param boolean $stared
     * @return State
     */
    public function setStared($stared)
    {
        $this->stared = $stared;

        return $this;
    }

    /**
     * Get stared
     *
     * @return boolean 
     */
    public function getStared()
    {
        return $this->stared;
    }

    /**
     * Set post
     *
     * @param \Cyniris\SyndicateBundle\Entity\Post $post
     * @return State
     */
    public function setPost(\Cyniris\SyndicateBundle\Entity\Post $post = null)
    {
        $this->post = $post;

        return $this;
    }

    /**
     * Get post
     *
     * @return \Cyniris\SyndicateBundle\Entity\Post 
     */
    public function getPost()
    {
        return $this->post;
    }

    /**
     * Set user
     *
     * @param \Cyniris\SyndicateBundle\Entity\User $user
     * @return State
     */
    public function setUser(\Cyniris\SyndicateBundle\Entity\User $user = null)
    {
        $this->user = $user;

        return $this;
    }

    /**
     * Get user
     *
     * @return \Cyniris\SyndicateBundle\Entity\User 
     */
    public function getUser()
    {
        return $this->user;
    }
}

File: src/main/java/com/hawkprime/syndicate/model/State.java
package com.hawkprime.syndicate.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import org.apache.commons.lang3.builder.ToStringBuilder;

/**
 * State Entity.
 *
 * @version 5.0.0
 * @author Juan D Frias <juandfrias@gmail.com>
 */
@Entity
@Table(name="post_state")
public class State {

	@Id
	@GeneratedValue
	private Long id;

	@Column(nullable=false)
	private Boolean read;

	@Column(nullable=false)
	private Boolean stared;

	@ManyToOne
	@JoinColumn(name="user_id", nullable=false)
	private User user;

	@ManyToOne
	@JoinColumn(name="post_id", nullable=false)
	private Post post;

	/**
	 * Gets the id.
	 *
	 * @return the id
	 */
	public Long getId() {
		return id;
	}

	/**
	 * Sets the id.
	 *
	 * @param id the id to set
	 */
	public void setId(final Long id) {
		this.id = id;
	}

	/**
	 * Checks if is read.
	 *
	 * @return the read
	 */
	public Boolean isRead() {
		return read;
	}

	/**
	 * Sets the read.
	 *
	 * @param read the read to set
	 */
	public void setRead(final Boolean read) {
		this.read = read;
	}

	/**
	 * Checks if is stared.
	 *
	 * @return the stared
	 */
	public Boolean isStared() {
		return stared;
	}

	/**
	 * Sets the stared.
	 *
	 * @param stared the stared to set
	 */
	public void setStared(final Boolean stared) {
		this.stared = stared;
	}

	/**
	 * Gets the user.
	 *
	 * @return the user
	 */
	public User getUser() {
		return user;
	}

	/**
	 * Sets the user.
	 *
	 * @param user the user to set
	 */
	public void setUser(final User user) {
		this.user = user;
	}

	/**
	 * Gets the post.
	 *
	 * @return the post
	 */
	public Post getPost() {
		return post;
	}

	/**
	 * Sets the post.
	 *
	 * @param post the post to set
	 */
	public void setPost(final Post post) {
		this.post = post;
	}

	/* (non-Javadoc)
	 * @see java.lang.Object#toString()
	 */
	@Override
	public String toString() {
		return new ToStringBuilder(this)
				.append("id", id)
				.append("read", read)
				.append("stared", stared)
				.append("user", user.getName())
				.append("post", post.getTitle())
				.toString();
	}
}
